src/uniprotkb/components/entry/Entry.tsx

```diff
@@ -220,6 +220,33 @@ const Entry: FC = () => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [dispatch, redirectedTo]);
 
+  const isObsolete = Boolean(
+    transformedData?.entryType === EntryType.INACTIVE &&
+      transformedData.inactiveReason
+  );
+
+  // Redirect to history when obsolete and not merged into a single new one
+  useEffect(() => {
+    if (
+      isObsolete &&
+      match?.params.accession &&
+      match?.params.subPage !== TabLocation.History
+    ) {
+      frame().then(() => {
+        history.replace(
+          getEntryPath(
+            Namespace.uniprotkb,
+            match?.params.accession,
+            TabLocation.History
+          )
+        );
+      });
+    }
+    // (I hope) I know what I'm doing here, I want to stick with whatever value
+    // match?.params.subPage had when the component was mounted.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dispatch, isObsolete]);
```

Comment:
```suggestion
  }, [isObsolete]);
```
`dispatch` not used by the `useEffect` so my thinking is it can be excluded